File: Instant_Problems/Strange_Array_Sort.cpp
#include<bits/stdc++.h>
using namespace std;

int main()
{
    int n;
    cin >> n;
    int* arr = new int[n];
    for(int i=0;i<n;i++)
    {
        cin >> arr[i];
    }
    sort(arr,arr+n);
    int* lower_arr = new int[n/2];
    int* upper_arr = new int[n/2];
    for(int i=0;i<n/2;i++)
    {
        lower_arr[i] = arr[i];
        upper_arr[i] = arr[i+(n/2)];
    }
    int* new_arr = new int[n];
    for(int i=0;i<n;i++)
    {
        if(i%2==0)
        {
            new_arr[i] = lower_arr[i/2];
        }
        else
        {
            new_arr[i] = upper_arr[i/2];
        }
    }
    for(int i=0;i<n;i++)
    {
        cout << new_arr[i] << " ";
    }
}

File: Instant_Problems/Array_Manipulation/Strange_Array_Sort.py
# taking array input from the user
arr=list(map(int,input("Please Enter the array : ").split()))

# sorting the array
arr.sort()

p = len(arr)
q = int((p/2))

# splitting the array in two parts, lower_arr having lower values 
# and upper_arr having upper values
lower_arr = [arr[i] for i in range(q)]
upper_arr = [arr[i+q] for i in range(q)]

# creating a new empty array
new_arr = [0] * p 

# main algorithm, merging lower_arr and upper_arr with some condition
for i in range(p):
    if(i%2==0):
        new_arr[i] = lower_arr[int(i/2)]
    else:
        new_arr[i] = upper_arr[int(i/2)]
        
# printing the arr
print(new_arr)
